internal/common/agent_util_unix.go

```diff
@@ -14,12 +14,18 @@ import (
 )
 
 const (
-	CatCommand       = "cat "
-	AppOwnerCommand  = "ps -u -p "
-	ConfigOutputPath = "/opt/aws/amazon-cloudwatch-agent/bin/config.json"
-	Namespace        = "CWAgent"
-	Host             = "host"
-	AgentLogFile     = "/opt/aws/amazon-cloudwatch-agent/logs/amazon-cloudwatch-agent.log"
+	CatCommand         = "cat "
+	AppOwnerCommand    = "ps -u -p "
+	ConfigOutputPath   = "/opt/aws/amazon-cloudwatch-agent/bin/config.json"
+	Namespace          = "CWAgent"
+	Host               = "host"
+	AgentLogFile       = "/opt/aws/amazon-cloudwatch-agent/logs/amazon-cloudwatch-agent.log"
+	installRetryNumber = 10
+	installRetryTime   = 30 * time.Second
+	debInstall         = "deb"
+	rpmInstall         = "rpm"
+	debInstallCommand  = "sudo dpkg -i -E ./amazon-cloudwatch-agent.deb"
+	rpmInstallCommand  = "sudo rpm -U ./amazon-cloudwatch-agent.rpm"
```

Comment:
This will ensure that we fail if the new version is not grater than the current version. Thus, we fail the canary deployment if you gave the wrong version number.